static/src/javascripts/projects/commercial/modules/hosted/onward.js

```diff
@@ -31,10 +33,12 @@ define([
                         new HostedCarousel.init();
                     });
                 })
-                .then(function () {
-                    performanceLogging.moduleEnd(moduleName);
-                });
+        } else {
+            fetchPromise = Promise.resolve();
         }
-        return Promise.resolve();
+
+        return fetchPromise.then(function () {
```

Comment:
yes, hosted gallery and hosted onward both need to return a promise to bootstrap. The promise will only be used to wait until all performance `moduleEnd` calls have been made. At that point, I can report the perf data.